src/cdomains/intDomain.ml

```diff
@@ -923,6 +923,461 @@ struct
   let project ik p t = t
 end
 
+module IntervalSetFunctor(Ints_t : IntOps.IntOps): S with type int_t = Ints_t.t and type t = (Ints_t.t * Ints_t.t) list =
+struct
+
+  module Interval_functor = IntervalFunctor(Ints_t) 
+
+  let name () = "IntervalSets"
+
+  type int_t = Ints_t.t
+
+  type t = (Ints_t.t * Ints_t.t) list [@@deriving eq, hash, ord]
+
+  let range ik = BatTuple.Tuple2.mapn Ints_t.of_bigint (Size.range ik)
+
+  let top () = failwith @@ "top () not implemented for " ^ (name ())
+
+  let top_of ik = [range ik]
+   
+  let bot () = []
+  
+  let bot_of ik = bot () 
+
+  let show (x: t) =
+    let show_interval i = Printf.sprintf "[%s, %s]" (Ints_t.to_string (fst i)) (Ints_t.to_string (snd i)) in
+    List.fold_left (fun acc i -> (show_interval i) :: acc) [] x |> List.rev |> String.concat ", " |> Printf.sprintf "[%s]"
+  
+  let cartesian_product l1 l2 = 
+    List.fold_left (fun acc1 e1 -> List.fold_left (fun acc2 e2 -> (e1, e2)::acc2) acc1 l2) [] l1
+
+  (* New type definition for the sweeping line algorithm used for implementiong join/meet functions. *)
+  type 'a event = Enter of 'a | Exit of 'a
+  
+  let unbox_event = function Enter x -> x | Exit x -> x
+
+  let _show_event = function 
+    | Enter x -> Printf.sprintf "Enter %s" (Ints_t.to_string x) 
+    | Exit x -> Printf.sprintf "Exit %s" (Ints_t.to_string x)
+    
+  let sort_events =
+    let cmp x y = 
+      let res = Ints_t.compare (unbox_event x) (unbox_event y) in
+      if res != 0 then res
+      else
+        begin
+          match (x, y) with
+          | (Enter _, Exit _) -> -1
+          | (Exit _, Enter _) -> 1
+          | (_, _) -> 0
+        end
+    in
+    List.sort cmp
+        
+  let interval_set_to_events (xs: t) = 
```

Comment:
Does this need to work on unordered intervals or are the argument interval sets guaranteed to be sorted?